src/Microsoft.AspNetCore.Server.IIS/HttpProtocol.cs

```diff
@@ -147,14 +167,27 @@ internal unsafe HttpProtocol(PipeFactory pipeFactory, IntPtr pHttpContext)
 
         public Stream RequestBody { get; set; }
         public Stream ResponseBody { get; set; }
-
         public IPipe Input { get; set; }
         public OutputProducer Output { get; set; }
 
         public IHeaderDictionary RequestHeaders { get; set; }
-        public IHeaderDictionary ResponseHeaders { get; set; } = new HeaderDictionary();
+        public IHeaderDictionary ResponseHeaders { get; set; }
         internal HttpApiTypes.HTTP_VERB KnownMethod { get; }
 
+        public int StatusCode
+        {
+            get { return _statusCode; }
+            set
+            {
+                // Http.Sys automatically sends 100 Continue responses when you read from the request body.
```

Comment:
this comment seems out of place. Did there use to be a check that prohibited 100 status codes and lower?